### ISSUE TITLE
Unable to use debug function in Arduino IDE 2.3.0

### ISSUE THREAD

**Issue by China-Mr-Hou** [stm32duino/Arduino_Core_STM32#2277]
**Describe the bug**
Recently I'm using Arduino IDE version 2.2.1, its debug function worked well on my H743VIT6, however after I upgrade to 2.3.0, it says I should select a programmer and doesn't show any more info. The “Tool" menu above doesn't have a programmer selection either. I tried copying the programmer.txt file included from Arduino GIGA R1 package from Arduino Official since GIGA board also uses STM32H7 as the core chip and it didn't change anything.

**To Reproduce**

1. Install Arduino IDE 2.3.0
2. Click on the Debug Tab one the left side
3. See error: No programmer selected for XXX

**Desktop (please complete the following information):**
 - OS: Windows 11 build 22621
 - Arduino IDE version: 2.3.0
 - STM32 core version: 2.7.1
 - Tools menu settings if not the default: Nothing changed.
 - Upload method: SWD

**Board (please complete the following information):**
 - Name: Generic H743VITx
 - Hardware Revision: None
 - Extra hardware used if any: None

**Additional context**
I have standalone OpenOCD and JLink software installed.

**Comment by fpistm** [comment]
Hi @China-Mr-Hou 
I saw this issue. It comes from changes from arduino-cli.
I've already started to work on this but need more time to properly handle that and I think I found some issues with the Arduino IDE 2

**Comment by China-Mr-Hou** [comment]
I wrote this comment after I have upgraded my IDE to 2.3.1 and found Arduino Team didn't fix this problem (since issue#2354 is intended). In my opinion, they may be thinking choosing a programmer instead of writing debug configuration json file is simpler. Anyway, I still appreciate the Arduino developing platform itself and every third party contributions for making hardware coding easy for everyone.
